userspace/libscap/scap.c

```diff
@@ -2717,29 +2718,81 @@ int32_t scap_disable_skb_capture(scap_t *handle)
 #endif
 }
 
+void delete_pid_vtid_map(scap_t *handle, uint64_t pid, uint64_t tid){
+	uint64_t vtid = get_tid_vtid_map(handle, tid);
+	if(vtid == 0){
+		return ;
+	}
+	uint64_t pid_vtid = pid<<32 | (vtid & 0xFFFFFFFF);
+	pid_vtid_info *pvi;
+	HASH_FIND_INT64(handle->m_pid_vtid_info, &pid_vtid, pvi);
+	if(pvi!=NULL){
+		HASH_DEL(handle->m_pid_vtid_info, pvi);
+		free(pvi);
+	}
+	delete_tid_vtid_map(handle, tid);
+}
+
+uint64_t get_pid_vtid_map(scap_t *handle, uint64_t pid, uint64_t vtid){
+	uint64_t pid_vtid = pid<<32 | (vtid & 0xFFFFFFFF);
+	int32_t uth_status = SCAP_SUCCESS;
+	pid_vtid_info *pvi;
+	HASH_FIND_INT64(handle->m_pid_vtid_info, &pid_vtid, pvi);
+	if(pvi!=NULL){
+		return pvi->tid;
+	}else{
+		return 0;
+	}
+}
+
 bool put_pid_vtid_map(scap_t *handle, uint64_t pid, uint64_t tid, uint64_t vtid){
 	int32_t uth_status = SCAP_SUCCESS;
 	pid_vtid_info *pvi;
-	HASH_FIND_INT64(handle->m_pid_vtid_info, &pid, pvi);
+	uint64_t pid_vtid = pid<<32 | (vtid & 0xFFFFFFFF);
+	HASH_FIND_INT64(handle->m_pid_vtid_info, &pid_vtid, pvi);
 	if(pvi==NULL){
 		pvi = (struct pid_vtid_info*)malloc(sizeof(pid_vtid_info));
-		pvi->pid_vtid = pid<<32 | (vtid & 0xFFFFFFFF);
+		pvi->pid_vtid = pid_vtid;
 		pvi->tid = tid;
 		uth_status = SCAP_SUCCESS;
 		HASH_ADD_INT64(handle->m_pid_vtid_info, pid_vtid, pvi);
 	}else {
 		pvi->tid = tid;
 	}
+	put_tid_vtid_map(handle, tid, vtid);
 	return true;
 }
 
-uint64_t get_pid_vtid_map(scap_t *handle, uint64_t pid, uint64_t vtid){
-	uint64_t pid_vtid = pid<<32 | (vtid & 0xFFFFFFFF);
+bool put_tid_vtid_map(scap_t *handle, uint64_t tid, uint64_t vtid){
 	int32_t uth_status = SCAP_SUCCESS;
```

Comment:
Could remove `uth_status` since it is not used? I've seen three warnings when compiling:
```
/kindling/probe/libs/agent-libs/userspace/libscap/scap.c: In function 'get_pid_vtid_map':                                                                                             
/kindling/probe/libs/agent-libs/userspace/libscap/scap.c:2738:10: warning: unused variable 'uth_status' [-Wunused-variable]                                                           
  int32_t uth_status = SCAP_SUCCESS;                                                       
          ^                                                                                                                                                                           
/kindling/probe/libs/agent-libs/userspace/libscap/scap.c: In function 'put_pid_vtid_map':                                                                                             
/kindling/probe/libs/agent-libs/userspace/libscap/scap.c:2749:10: warning: variable 'uth_status' set but not used [-Wunused-but-set-variable]                                         
  int32_t uth_status = SCAP_SUCCESS;                                                                                                                                                  
          ^                                                                                                                                                                           
/kindling/probe/libs/agent-libs/userspace/libscap/scap.c: In function 'put_tid_vtid_map':                                                                                             
/kindling/probe/libs/agent-libs/userspace/libscap/scap.c:2767:10: warning: variable 'uth_status' set but not used [-Wunused-but-set-variable]                                         
  int32_t uth_status = SCAP_SUCCESS;  
          ^      
```

Reply:
Forget it. It can't be compiled if removed.